### PR TITLE
Add image manager plugin API

### PR DESCRIPTION
**Pull request by IntelOrca**
Specifically for the devtools plugin to examine the image list and fragmentation of it.

A lot of changes, but most just because I refactored the Image.cpp definitions into its own header. `Drawing.h` needs trimming, particularly as changes to it cause over 500 source files to need recompiling.

### CODE REVIEW

File: src/openrct2-ui/scripting/ScImageManager.hpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2020 OpenRCT2 developers
```
Comment (tupaschoal) [inline src/openrct2-ui/scripting/ScImageManager.hpp:2]:
```suggestion
 * Copyright (c) 2014-2021 OpenRCT2 developers
```

File: src/openrct2-ui/scripting/ScImageManager.hpp [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        DukValue getAvailableAllocationRanges() const
+        {
+            auto ranges = GetAvailableAllocationRanges();
+            duk_push_array(_ctx);
+            duk_uarridx_t index = 0;
+            for (const auto& range : ranges)
+            {
+                auto value = CreateImageIndexRange(range.BaseId, range.Count);
+                value.push();
+                duk_put_prop_index(_ctx, -2, index);
```
Comment (tupaschoal) [inline src/openrct2-ui/scripting/ScImageManager.hpp:71]: What is `-2`? Sounds like it's worth adding a comment

File: src/openrct2/drawing/Image.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2020 OpenRCT2 developers
```
Comment (tupaschoal) [inline src/openrct2/drawing/Image.h:2]:
```suggestion
 * Copyright (c) 2014-2021 OpenRCT2 developers
```

**Review by Gymnasiast — COMMENTED** [review]
AFAIC, this can be merged once Tulio’s line notes are sorted.

**Review by tupaschoal — APPROVED** [review]

**Pull request merged**